spec/unit/knife/core/windows_bootstrap_context_spec.rb

```diff
@@ -96,74 +86,74 @@
   describe "#get_log_location" do
 
     context "when config_log_location value is nil" do
+      let(:chef_config) { { config_log_location: nil } }
       it "sets STDOUT in client.rb as default" do
-        bootstrap_context.instance_variable_set(:@chef_config, Mash.new(config_log_location: nil))
         expect(bootstrap_context.get_log_location).to eq("STDOUT\n")
       end
     end
 
     context "when config_log_location value is empty" do
+      let(:chef_config) { { config_log_location: "" } }
       it "sets STDOUT in client.rb as default" do
-        bootstrap_context.instance_variable_set(:@chef_config, Mash.new(config_log_location: ""))
         expect(bootstrap_context.get_log_location).to eq("STDOUT\n")
       end
     end
 
     context "when config_log_location value is STDOUT" do
+      let(:chef_config) { { config_log_location: STDOUT } }
       it "sets STDOUT in client.rb" do
-        bootstrap_context.instance_variable_set(:@chef_config, Mash.new(config_log_location: STDOUT))
         expect(bootstrap_context.get_log_location).to eq("STDOUT\n")
       end
     end
 
     context "when config_log_location value is STDERR" do
+      let(:chef_config) { { config_log_location: STDERR } }
       it "sets STDERR in client.rb" do
-        bootstrap_context.instance_variable_set(:@chef_config, Mash.new(config_log_location: STDERR))
         expect(bootstrap_context.get_log_location).to eq("STDERR\n")
       end
     end
 
     context "when config_log_location value is path to a file" do
+      let(:chef_config) { { config_log_location: "C:\\chef\\chef.log" } }
       it "sets file path in client.rb" do
-        bootstrap_context.instance_variable_set(:@chef_config, Mash.new(config_log_location: "C:\\chef\\chef.log"))
```

Comment:
Yeah so this is awful, but its a symptom of excessive stubbing, the chef_config should just be Chef::Config.

That is a global variable so has all the global variable issues associated with testing but we're already calling Chef::Config.reset! in the spec_helper on every example to clear the global state, so we don't have to overly mock the Chef::Config.

Reply:
I'm also less and less concerned about let()'ing and before()'ing everything:

```ruby
it "sets file path in client.rb" do
  bootstrap_context.chef_config.config_log_location "C:\\chef\\chef.log"
  [....]
end
```

Reply:
I had guessed it was because at one point `@chef_config` was not as easily injectable. With the initializer taking this as a param, and this not being used to avoid a weird side effect or something, it seems pretty pointless to me.

I do agree with you that setting the specific value a test is concerned with on the default config object makes more sense as a realistic testing scenario though, I can change these.

Reply:
i would guess it was written this way just because someone thought that in unit tests you had to mock everything.